hail/python/hail/methods/impex.py

```diff
@@ -1824,31 +1872,61 @@ def import_vcf(path,
     return MatrixTable(jmt)
 
 
-@typecheck(path=oneof(str, sequenceof(str)))
-def index_bgen(path):
+@typecheck(path=oneof(str, sequenceof(str)),
+           directory=nullable(str),
```

Comment:
I thought about doing that, but was concerned that since we rely on the extension being `.idx2` in LoadBgen, I didn't want to make the user specify `.idx2` as the extension . I could check the index file name the user provides ends in `.idx2`. I also thought the directory interface was easier when using globs and indexing multiple files at the same time.

Reply:
I thought about this more, and decided I like your interface better.

Reply:
There's one problem with my interface: you can't run getFileStatus on the key of the index file map (since it probably doesn't exist yet) so you won't get the same normalization as with import_bgen which is unfortunate.

Actually, instead of calling getFileStatus, you might try makeQualified.  That will add the scheme and make relative path absolute but looks like it doesn't need the file to exist.

Reply:
The BGEN file has to exist already and is checked in a different function.

Reply:
Oh, right, we're just normalizing the bgen location.

Reply:
Looks good.